### PR TITLE
Direct git to ignore commonly untracked Python and Jetbrains files.

### PR DESCRIPTION
**Pull request by MichaelWakahe**
This is a .gitignore file with files that should not be tracked by git.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+nosetests.xml
+coverage.xml
+*.cover
+*.py,cover
+.hypothesis/
+.pytest_cache/
+cover/
+
+# Sphinx documentation
+docs/_build/
```
Comment (itsjohnward) [inline .gitignore:56]: We have this in `docs/.gitignore` currently. Can remove one or the other

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,123 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
```
Comment (itsjohnward) [inline .gitignore:3]: Ignoring `.pyc` in api/.gitignore currently. Should be able to remove that file

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+# Pyre type checker
+.pyre/
+
+# pytype static type analyzer
+.pytype/
+
+
+# IntelliJ
+*.iml
+.idea/
```
Comment (itsjohnward) [inline .gitignore:123]: Might as well add these for VSCode as well:

```py
.vscode/*
*.code-workspace
```

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+# Spyder project settings
+.spyderproject
+.spyproject
+
+# Rope project settings
+.ropeproject
+
+# mkdocs documentation
+/site
```
Comment (itsjohnward) [inline .gitignore:107]: We are using sphinx, rather than mkdocs, so this one can be excluded

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+# mypy
+.mypy_cache/
+.dmypy.json
+dmypy.json
+
+# Pyre type checker
+.pyre/
+
+# pytype static type analyzer
+.pytype/
```
Comment (itsjohnward) [inline .gitignore:118]: I would like to add mypy at some point, but currently we're not using any of these type checkers (mypy, pyre, pytype). So I think we can exclude them until it's relevant

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+ENV/
+env.bak/
+venv.bak/
+
+# Spyder project settings
+.spyderproject
+.spyproject
+
+# Rope project settings
+.ropeproject
```
Comment (itsjohnward) [inline .gitignore:104]: Not using rope currently

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+#   having no cross-platform support, pipenv may install dependencies that don't work, or not
+#   install all needed dependencies.
+#Pipfile.lock
+
+# PEP 582; used by e.g. github.com/David-OConnor/pyflow
+__pypackages__/
+
+# Celery stuff
+celerybeat-schedule
+celerybeat.pid
```
Comment (itsjohnward) [inline .gitignore:85]: Nothing celery-related in this repo. Can use the API client in a celery task, but scheduling and all that would not go in this package. Can exclude

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
+
+# pipenv
+#   According to pypa/pipenv#598, it is recommended to include Pipfile.lock in version control.
+#   However, in case of collaboration, if having platform-specific dependencies or dependencies
+#   having no cross-platform support, pipenv may install dependencies that don't work, or not
+#   install all needed dependencies.
+#Pipfile.lock
```
Comment (itsjohnward) [inline .gitignore:78]: We can uncomment to ignore Pipfile.lock.
You want it committed in an *application*, like what we deploy to our servers, but not in an installable package. Here, we want requirements to be as permissive as possible so it doesn't cause unnecessary conflicts in installers' environments.

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
```
Comment (itsjohnward) [inline .gitignore:71]: Can uncomment to ignore. We support many different python versions - shouldn't need to specify a specific one with pyenv

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+*.py,cover
+.hypothesis/
+.pytest_cache/
+cover/
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+target/
```
Comment (itsjohnward) [inline .gitignore:59]: Not using PyBuilder - can remove

**Review by itsjohnward — CHANGES_REQUESTED** [review]
Looks all right, but I think it's somewhat overkill - left some comments on irrelevant lines so we can slim it down a bit to what's really necessary.

Also, let's consolidate `api/.gitignore`, `docs/.gitignore`, and `r-client/.gitignore` up into this one

**Comment by jli** [comment]
the major parts of this were added in #264, thanks @MichaelWakahe !

**Pull request closed without merging**